src/types.rs

```diff
@@ -60,15 +63,15 @@ impl TypeRef {
     }
 }
 
-/// Represents a [CoreISLTypeDefinition] which stoes a resolved core ISl type using [TypeStore]
+/// Represents a [BuiltInTypeDefinition] which stores a resolved builtin ISl type using [TypeStore]
 #[derive(Debug, Clone, PartialEq)]
-pub enum CoreISLTypeDefinition {
-    IonType(IonType),
-    Other(TypeDefinitionImpl),
+pub enum BuiltInTypeDefinition {
+    Atomic(IonType, bool), // bool represents if the atomic ion type is nullable or not
```

Comment:
Since the `bool` often requires an explanatory comment, I wonder if it's worth instead creating an enum:

```rust
enum Nullability {
  Nullable,
  NotNullable
}
```

to make it a bit more obvious? (The `Reader` does this sort of thing in `ion-rust` and I keep meaning to find another representation for it 😞)